docs/src/main/asciidoc/building-my-first-extension-guide.adoc

```diff
@@ -0,0 +1,272 @@
+
+At this point, you should have understand that most of the magic will happen at the Augmentation Build time thanks to the deployment module.
+
+Quarkus Application Bootstrap
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There are three distinct bootstrap phases of a Quarkus application.
+
+* Augmentation: During the build time, the Quarkus extensions will load and scan your application's bytecode (including the dependencies) and configuration. Once all the metadata collected, the extensions can pre-process the frameworks bootstrap actions like your ORM, DI or REST controllers configurations. The result of the bootstrap is directly recorded into bytecode and will be part of your final application package.
+
+* Static Init: During the run time, Quarkus will execute first a static init method which contains some extensions actions/configurations. When you will do your native packaging, this static method will be pre-process during the build time and the eventual generated objects will be serialized in the final native executable, so this code will not be executed in a native node (Imagine you execute a Fibonacci function during this phase, the result of the computation will be directly recorded in the native executable). In a JVM mode, there is no much difference.
+
+* Runtime Init: Well nothing fancy here, we do classic run time code executation. So, the more code you run on the 2 above phases, the faster your application will be.
+
+Now that everything is explained, we can start coding !!
+
+Maven setup
+~~~~~~~~~~~
+
+=== The parent module
+At the moment, all the extensions are hosted in the https://github.com/quarkusio/quarkus[Quarkus repository].
+
+[source, shell]
+---------------
+$ git clone https://github.com/quarkusio/quarkus.git
+$ cd ./quarkus/extensions
+$ mkdir smallrye-health
+---------------
+
+Your extension should be a multi-module project. So let's start by the parent `pom.xml`.
+[source, xml]
+---------------
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>quarkus-build-parent</artifactId>
+        <groupId>io.quarkus</groupId>
+        <version>999-SNAPSHOT</version>
+        <relativePath>../../build-parent/pom.xml</relativePath>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>quarkus-smallrye-health-parent</artifactId>
+    <name>Quarkus - SmallRye Health</name>
+
+    <packaging>pom</packaging>
+    <modules>
+        <module>deployment</module>
+        <module>runtime</module>
+    </modules>
+</project>
+---------------
+
+Your extensions is a child project of `quarkus-build-parent` and declares 2 sub-modules deployment and runtime. It's time now to create them.
```

Comment:
extensions -> extension

It could be mentioned that it doesn't have to have `quarkus-build-parent` as the parent.